### DISCUSSION TITLE
[Request] Add a status badge for a PR that targets a particular branch (e.g. main)

### DISCUSSION THREAD

**Discussion by danielfdickinson** [community/community#9091] [Actions]
As suggested by @ethomson in response to https://github.com/github/docs/pull/12331 (https://github.com/github/docs/pull/12331#issuecomment-989116954) I'm adding feedback on why I'd like to see status badge for a PR that targets a particular branch.

Basically I want to use the status badge on deploy previews of the site or docs (depending on if the preview is for a website or for documentation of/about code) for an at-a-glance view of whether the proposed PR is 'CI green' if it were merged into the target (e.g. 'main').

That is what I thought the ``pull_request`` 'event' was for in the status badge documentation. I'm actually not sure what the purpose of the existing option (status of a PR _from_ a particular branch) is.

**Comment by ethomson** [comment]
Thanks @danielfdickinson  - I don't think that there's really a way to capture this today, but I think that we can indeed add this support.  Just to help understand, are you using this to build a dashboard or something along those lines?

I have a PR open to update the documentation to stop using `pull_request` as the example, I agree that it's misleading.

**Reply by danielfdickinson** [reply]
I wasn't so ambitious in my thinking although I actually love the idea of being able to use a static site generator to create a dashboard of a set of my repositories. Thank you for the idea!

I was actually thinking more in terms of using on a per-repo website/docsite basis, but I like the dashboard idea better. I'll have to think some more about what I'd want to see (obviously there are a 'bazillion' badges one could include from various sources, it's figure out what is relevant for a particular purpose-oriented dashboard.

In term of pull requests, an even better option would be the number of pull requests against (e.g.) 'main' that are green and the number that are 'red'. Along with the current status of significant branches like 'main' or 'staging' one could have a static page that gives an at-a-glance view of a project (one or more repos, depending on the project).

I hope that makes some kind of sense outside my brain!

**Comment by HariSekhon** [comment]
IMO the behaviour of the `badge.svg?event=pull_request` is broken and inconsistent with workflow query `event: pull_request`.

It should filter for the result of the latest pull_request triggered workflow in a similar fashion to what is seen from `.../workflows/file.yaml?query=event%3Apull_request`

See here for a simple example of the inconsistency of this behaviour:

https://github.com/HariSekhon/GitHub-Actions-Contexts

The pull request badge there has `no status` but if you click through then it clearly has a run and status.
